### PR TITLE
feat: ability to download all the 2FA recovery codes

### LINKED ISSUE #938: feat: Ability to download all the 2FA recovery codes
### Feature Description

Add a download button to the recovery code dialog to allow users to download all the codes locally in either .txt or .csv format. Currently, users have to copy each code individually and save it somewhere in order to store.

### Use Case

_No response_

### Proposed Solution

_No response_

### Alternatives (optional)

_No response_

### Additional Context

_No response_

### Please check the boxes that apply to this feature request.

- [ ] I have searched the existing feature requests to make sure this is not a duplicate.
- [ ] I have provided a detailed description of the requested feature.
- [ ] I have explained the use case or scenario for this feature.
- [ ] I have included any relevant technical details or design suggestions.
- [ ] I understand that this is a suggestion and that there is no guarantee of implementation.
- [x] I want to work on creating a PR for this issue if approved

### PR DESCRIPTION
**Pull request by anikdhabal**
fixes #938 

https://github.com/documenso/documenso/assets/81948346/8bf807ae-41b0-417b-9f55-d96584e71acb

### CODE REVIEW

File: apps/web/src/components/forms/2fa/view-recovery-codes-dialog.tsx [hunk 1]
```diff
@@ -139,8 +150,11 @@ export const ViewRecoveryCodesDialog = ({ open, onOpenChange }: ViewRecoveryCode
                 <RecoveryCodeList recoveryCodes={viewRecoveryCodesData.recoveryCodes} />
               )}
 
-              <div className="mt-4 flex flex-row-reverse items-center justify-between">
+              <div className="mt-4 flex flex-row-reverse items-center gap-2">
                 <Button onClick={() => onOpenChange(false)}>Complete</Button>
+                <a download="documenso-2FA-recovery-codes.txt" href={recoveryCodesUrl}>
+                  <Button>Download</Button>
```
Comment (dephraiim) [inline apps/web/src/components/forms/2fa/view-recovery-codes-dialog.tsx:156]: Can this button be a secondary button?

**Review by dephraiim — CHANGES_REQUESTED** [review]

File: apps/web/src/components/forms/2fa/view-recovery-codes-dialog.tsx (hunk 1 above)
Comment (anikdhabal) [reply apps/web/src/components/forms/2fa/view-recovery-codes-dialog.tsx:156]: Yeah, sure.

**Review by sachin-duhan — COMMENTED** [review]
It's normally recommended to create a utility file that handles Blob / download functionality. Creating or pre-processing file information may not be a part of useEffect.

File: apps/web/src/components/forms/2fa/view-recovery-codes-dialog.tsx [hunk 2]
```diff
@@ -62,6 +63,16 @@ export const ViewRecoveryCodesDialog = ({ open, onOpenChange }: ViewRecoveryCode
     return 'view';
   }, [viewRecoveryCodesData, isViewRecoveryCodesSubmitting]);
 
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (anikdhabal) [reply apps/web/src/components/forms/2fa/view-recovery-codes-dialog.tsx:66]: Thanks for the review. Yeah, it can be a function.

**Review by Mythie — APPROVED** [review]

**Comment by Mythie** [comment]
LGTM! Thanks for the improvement 🎉

**Comment by Mythie** [comment]
/tip 200 

Thanks for all your recent PR's while we might be slow to review and address them we do appreciate all the work!

**Comment by anikdhabal** [comment]
> /tip 200
> 
> Thanks for all your recent PR's while we might be slow to review and address them we do appreciate all the work!

Thank you, Lucas. It means a lot.

**Pull request merged**